Modules/feed/feed_model.php

```diff
@@ -516,7 +516,12 @@ public function get_timevalue($id)
         if ($engine == Engine::VIRTUALFEED) { //if virtual get it now
             $this->log->info("get_timevalue() calling VIRTUAL lastvalue $id");
             $lastvirtual = $this->EngineClass(Engine::VIRTUALFEED)->lastvalue($id);
-            return array('time'=>$lastvirtual['time'], 'value'=>$lastvirtual['value']);
+            if ($lastvirtual) {
+                $timevalue = array('time'=>$lastvirtual['time'], 'value'=>$lastvirtual['value']);
+            } else {
+                $timevalue = array('time'=>null, 'value'=>null);
```

Comment:
What problem is this fixing?

Reply:
has hit errors when $lastvirtual is null and has no ['time'] or ['value']. so I checked for that and returned null values for the `time` and `value`.

Reply:
**Time** must not be null but **value** can be null.
This fix should go to where the original error came from, the engine class lastvalue() function.